server/src/models/people/models.js

```diff
@@ -1,69 +1,58 @@
-import { camel_case_headcount_model_names } from "./headcount_model_utils.js";
-
-export default function define_resource_models(model_singleton) {
-  const _headcount_records_by_model_and_dept_code = {};
-
-  class HeadcountModel {
-    constructor(obj) {
-      Object.assign(this, obj);
-    }
-    static model_name() {
-      throw new Error(`
-        Error: bad OOP alert, don't use HeadcountModels class directly.
-        Need to create a new class extending it that overwrites the static 
-        model_name() function with one that returns the name of the actual model.
-      `);
-    }
-    static get_model_name() {
-      try {
-        return this.model_name();
-      } catch (e) {
-        console.log(e);
-      }
-    }
-    static register(obj) {
-      const { dept_code } = obj;
-
-      const inst = new HeadcountModel(obj);
-
-      const model_name = this.get_model_name();
-
-      if (!_headcount_records_by_model_and_dept_code[model_name]) {
-        _headcount_records_by_model_and_dept_code[model_name] = {};
-      }
-
-      if (!_headcount_records_by_model_and_dept_code[model_name][dept_code]) {
-        _headcount_records_by_model_and_dept_code[model_name][dept_code] = [];
-      }
-
-      _headcount_records_by_model_and_dept_code[model_name][dept_code].push(
-        inst
-      );
-    }
-    static get_dept_records(dept_code) {
-      const model_name = this.get_model_name();
-      return (
-        _headcount_records_by_model_and_dept_code[model_name][dept_code] || []
-      );
-    }
-    static get_gov_records() {
-      const model_name = this.get_model_name();
-      return (
-        _headcount_records_by_model_and_dept_code[model_name]["ZGOC"] || []
-      );
-    }
-  }
-
-  const headcount_model_factory = (model_name) => {
-    const ExtendedHeadcountModel = class extends HeadcountModel {
-      static model_name() {
-        return model_name;
-      }
-    };
-    model_singleton.define(model_name, ExtendedHeadcountModel);
-  };
-
-  camel_case_headcount_model_names.forEach((model_name) =>
-    headcount_model_factory(model_name)
+import _ from "lodash";
+import mongoose from "mongoose";
+
+import {
+  create_resource_by_id_attr_dataloader,
+  create_resource_by_foreignkey_attr_dataloader,
+} from "../loader_utils.js";
+import {
+  pkey_type,
+  parent_fkey_type,
+  fyear_type,
+  number_type,
+  str_type,
+  bilingual,
+} from "../model_utils.js";
+
+export default function (model_singleton) {
+  const EmployeeWholeDataSchema = mongoose.Schema({
+    dept_code: str_type,
```

Comment:
Follow up, more of a next-steps thing once these by-department models are worked out, note that by cases above didn't cover the government level infographic. Generally, we do not want to make the gov level infographic load all the granular department data! What we tend to do is pre-calculate the government level totals separately and store them in their own mongo collection(s). Preferably all in one, since we _do_ have a limit on total number of collections, and burning them for a single document each would be a waste.

In some cases, the gov level documents may look exactly the same as the dept level documents, just without a `dept_code` field. In other cases, the gov level data may need additional pre-calculated stats. It depends on what data the current corresponding infographics uses. In the people data case, the gov panels do _not_ (currently) require any additional stats. Many of the financial panels, on the other hand, do (as they contain references to things like which department is the largest spender in addition to plain government totals; don't want to load all of the data just to calculate that client side, might as well re-calculate it server side). This is a big part of what I referred to when I said there's going to be feedback loops between the server side stuff and the client side stuff.

Note: we're free to save a lot of pre-calculated (derived) data in our databases because all our data is re-populated on every deploy and read-only in between them :tada:. Lots of optimizations we can make because of that. The schema design would be VERY different if this data was read-write, since storing duplicate/derived values in that case would just be asking for data inconsistencies.